XFBindableStackLayout/Controls/BindableStackLayout.cs

```diff
@@ -1,55 +1,134 @@
         {
             get { return (string)GetValue(TitleProperty); }
             set { SetValue(TitleProperty, value); }
         }
-        public static readonly BindableProperty TitleProperty =
-            BindableProperty.Create(nameof(Title), typeof(string), typeof(BindableStackLayout),
-                                    propertyChanged: (bindable, oldValue, newValue) => ((BindableStackLayout)bindable).PopulateHeader());
+        public static readonly BindableProperty TitleProperty = BindableProperty.Create(
+            nameof(Title), typeof(string),
+            typeof(BindableStackLayout),
+            propertyChanged: (bindable, oldValue, newValue) => ((BindableStackLayout)bindable).PopulateHeader()
+        );
+
+        public object SelectedItem
+        {
+            get { return GetValue(SelectedItemProperty); }
+            set { SetValue(SelectedItemProperty, value); }
+        }
+
+        public static BindableProperty SelectedItemProperty = BindableProperty.Create(
+            propertyName: "SelectedItem",
+            returnType: typeof(object),
+            declaringType: typeof(BindableStackLayout),
+            defaultValue: null,
+            defaultBindingMode: BindingMode.OneWay,
+            propertyChanged: OnSelectedItemChanged
+        );
+
+        private static void OnSelectedItemChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var itemsView = (BindableStackLayout)bindable;
+            if (newValue == oldValue)
+                return;
+
+            itemsView.SetSelectedItem(newValue);
+        }
+
+        protected virtual void SetSelectedItem(object selectedItem)
+        {
+            var handler = SelectedItemChanged;
+            if (handler != null)
+                handler(this, new SelectedItemChangedEventArgs(selectedItem));
+        }
 
         void PopulateItems()
         {
             if (ItemsSource == null) return;
             foreach (var item in ItemsSource)
             {
-                var itemTemplate = ItemDataTemplate.CreateContent() as View;
-                itemTemplate.BindingContext = item;
-                Children.Add(itemTemplate);
+                Children.Add(GetItemView(item));
             }
         }
 
         void PopulateHeader() => header.Text = Title;
+
+        protected virtual View GetItemView(object item)
+        {
+            var content = ItemDataTemplate.CreateContent();
+
+            var view = content as View;
+            if (view == null)
+                return null;
+
+            view.BindingContext = item;
+
+            var gesture = new TapGestureRecognizer
+            {
+                Command = ItemSelectedCommand,
+                CommandParameter = item
+            };
+
+            AddGesture(view, gesture);
+
+            return view;
+        }
+
+        protected readonly ICommand ItemSelectedCommand;
+
+        protected void AddGesture(View view, TapGestureRecognizer gesture)
```

Comment:
1. Why `protected`? In my opinion should be `private`.
2.  There are different types of Gestures, it will make sense to rename the method to `AddTapGestureRecognize`.


P.S.: In order to keep the code style consistent the access modifier `private` should be removed as it is a default access modifier.